dandi/cli/cmd_validate.py

```diff
@@ -30,12 +31,38 @@ def validate_bids(
     if report_flag and not report:
         report = report_flag
 
-    validate_bids_(
+    validation_result = validate_bids_(
         *paths,
         report=report,
         schema_version=schema,
         devel_debug=devel_debug,
     )
+    missing_files = []
+    for pattern in validation_result["schema_tracking"]:
+        if pattern["mandatory"]:
+            missing_files.append(pattern["regex"])
```

Comment:
Python comprehensions make code more concise and readable.  Here it could be

```suggestion
    missing_files = [
        pattern["regex"]
        for pattern in validation_result["schema_tracking"]
        if pattern["mandatory"]
    ]
```
please RF similarly the one below following this pattern.